eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1062,7 +1162,7 @@ public function countInstancesOfType($typeId)
      * @param mixed $typeId
      * @param int $status
      */
-    public function deleteFieldDefinitionsForType($typeId, $status)
+    public function deleteFieldDefinitionsForType($typeId, $status, array $fieldDefinitions = [])
```

Comment:
Missing PhpDoc, missing abstract Gateway method prototype update.
But... TBH this is unnecessary redundancy, Field Definition Ids can be inferred from `$typeId` and `$status`.
Delete query should be based on parent table, ideally avoiding `WHERE ... IN` in favor of `WHERE EXISTS (...))`.